spec/std/hash_spec.cr

```diff
@@ -955,6 +977,16 @@ describe "Hash" do
       h2.should eq({"1a" => "a", "2b" => "b", "3c" => "c"})
     end
 
+    it "does not retain default value" do
+      h = Hash(Int32, String).new("a")
+      expect_raises(KeyError, "Missing hash key: 0") { h.transform_keys(&.succ)[0] }
+    end
+
+    it "does not retain default block" do
+      h = Hash(Int32, String).new { |_, _| "b" }
+      expect_raises(KeyError, "Missing hash key: 0") { h.transform_keys(&.succ)[0] }
+    end
+
```

Comment:
Wondering if it's more readable to check that `#[]?` returns `nil` instead of `#[]` raising exception.

Reply:
Done